Source/JavaScriptCore/builtins/JSIteratorPrototype.js

```diff
@@ -310,3 +310,93 @@ function reduce(reducer /*, initialValue */)
 
     return accumulator;
 }
+
+// https://tc39.es/proposal-iterator-chunking/#sec-iterator.prototype.chunks
+function chunks(chunkSize)
+{
+    "use strict";
+
+    if (!@isObject(this))
+        @throwTypeError("Iterator.prototype.chunks requires that |this| be an Object.");
+
+    var numChunkSize = @toNumber(chunkSize);
+    if (numChunkSize !== numChunkSize)
+        @throwRangeError("Iterator.prototype.chunks requires that argument not be NaN.");
+
+    var intChunkSize = @toIntegerOrInfinity(numChunkSize);
+    if (intChunkSize < 1 || intChunkSize > 0xffffffff)
```

Comment:
Would be nice to either utilize `@MAX_ARRAY_INDEX` somehow or add a new constant.